packages/host/app/services/matrix-service.ts

```diff
@@ -377,114 +379,149 @@ export default class MatrixService extends Service {
     body: string | undefined,
     attachedCards: CardDef[] = [],
     context?: OperatorModeContext,
+    clientGeneratedId?: string, // used to resend failed messages
```

Comment:
why is this optional?

Reply:
clientGeneratedId is generated in `setPendingMessage` and so for non-"resend" scenario, clientGeneratedId will be undefined.